README.md

```diff
@@ -1,2 +1,431 @@
+  - Frameworks (e.g. Redux)
+  - Utility functions/constants
+  - Reducers/Synchronizers
+  - Actions
+  - Components
+
+* When there are more than 5 import statements, add blank lines between groups.
+
+  ```js
+  import { connect } from 'react-redux';
+  import { Link } from 'react-router-dom';
+  import { Card, Button } from 'semantic-ui-react';
+  import { fetchUsers } from 'actions/users';
+  import searchBy from 'utils/search';
+
+  import Search from 'components/helpers/Search';
+  import Page from 'components/layouts/Page';
+  ```
+
+## JSX
+
+JSX leverages syntactic similarity with HTML so similar conventions should be observed.
+
+* rendering
+  ```js
+  // good
+  render() {
+    return <Button text="Save" onChange={this.save}/>;
+  }
+  // good
+  render() {
+    return (
+      <div className="action-bar">
+        <Button text="Save" onChange={this.save}/>
+        <Button text="Cancel" onChange={this.goBack}/>
+      </div>
+    );
+  }
+
+  // bad
+  render() {
+    return <div className="action-bar">
+      <Button text="Save" onChange={this.save}/>
+      <Button text="Cancel" onChange={this.goBack}/>
+    </div>;
+  }
+  ```
+
+* use double quotes for attributes
+  ```js
+  // good
+  <Button text="Save" onChange={this.save}/>
+
+  // bad
+  <Button text='Save' onChange={this.save}/>
+  ```
+
+* Do not use braces around double quotes
+  ```js
+  // good
+  <Button text="Save" onChange={this.save}/>
+
+  // bad
+  <Button text={"Save"} onChange={this.save}/>
+  ```
+
+* Do not use spaces around =
+  ```js
+  // good
+  <Button text="Save" onChange={this.save}/>
+
+  // bad
+  <Button text = "Save" onChange = {this.save}/>
+  ```
+
+* Inline short components
+  ```js
+  // good
+  <Button text="Save" onChange={this.save}/>
+
+  // bad
+  <Button
+    text="Save"
+    onChange={this.save}/>
+  ```
+
+* Multiline each prop on it's own line
+  ```js
+  // good
+  <Input
+    type="email"
+    label="Email"
+    placeholder="name@example.com"
+    onChange={this.setValue('email')}
+    onClick={this.selectText}/>
+
+  // bad
+  <Input
+    type="email" label="Email" placeholder="name@example.com"
+    onChange={this.setValue('email')} onClick={this.selectText}/>
+  ```
+
+* Keys should come first and may be on the first line when multilined.
+  ```js
+  // good
+  <ul>
+    {cars.map((car) => (
+      <Car key={car.id}
+        car={car}
+        detail="list"
+        showPrice={true}/>
+    ))}
+  </ul>
+
+  // bad
+  <ul>
+    {cars.map((car) => (
+      <Car
+        car={car}
+        detail="list"
+        key={car.id}
+        showPrice={true}/>
+    ))}
+  </ul>
+  ```
+
+* Multiline contents if there are nested elements.
+  ```js
+  // good
+  <p>Hi {name}, welcome to the site!</p>
+  // good
+  <p>
+    <i className="icon-checkmark"/>
+    <strong>Hi {name}</strong>, welcome to the site!
+  </p>
+
+  // bad
+  <p><i className="icon-checkmark"/><strong>Hi {name}</strong>, welcome to the site!</p>
```

Comment:
So multiline even it it is shorter version i.e. `<p><i className="icon-checkmark"/>Agree</p>`?

Reply:
That's right on the edge. I could be persuaded either way